nifi-nar-bundles/nifi-marklogic-bundle/nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/ExtensionCallMarkLogic.java

```diff
@@ -176,28 +174,54 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
             } else if (!requiresInput) {
                 flowFile = session.create();
             }
-            RequestParameters requestParameters = new RequestParameters();
-            String paramPrefix = "param";
-            List<PropertyDescriptor> parameterProperties = propertiesByPrefix.get(paramPrefix);
-            if (parameterProperties != null) {
-                for (final PropertyDescriptor propertyDesc: parameterProperties) {
-                    String paramName = propertyDesc.getName().substring(paramPrefix.length() + 1);
-                    String paramValue = context.getProperty(propertyDesc).evaluateAttributeExpressions(flowFile).getValue();
-                    PropertyValue separatorProperty = context.getProperty("separator:"+propertyDesc.getName());
-                    if (separatorProperty != null && separatorProperty.getValue() != null && !separatorProperty.getValue().isEmpty()) {
-                        requestParameters.add(
-                            paramName,
-                            paramValue.split(Pattern.quote(separatorProperty.evaluateAttributeExpressions(flowFile).getValue()))
-                        );
-                    } else {
-                        requestParameters.add(paramName, paramValue);
-                    }
+
+            RequestParameters requestParameters = buildRequestParameters(context, flowFile);
+            BytesHandle bytesHandle = buildWriteHandle(context, session, flowFile);
+
+            String method = context.getProperty(METHOD_TYPE).getValue();
+            ServiceResultIterator resultIterator = resourceManager.callService(method, bytesHandle, requestParameters);
+
+            while(resultIterator.hasNext()) {
+                ServiceResult result = resultIterator.next();
+                session.append(flowFile, out -> out.write(result.getContent(new BytesHandle()).get()));
+            }
+
+            synchronized(session) {
```

Comment:
This is the main fix - the synchronized block is only around the transfer/commit. It was previously including the call to MarkLogic. 

I actually don't know if the synchronized block is needed or not - I know it's in several of our processors - seems worth moving to a protected method in the abstract parent class and adding a comment as to why it's needed.